### PR TITLE
Store: Remove remaining dependencies on old countries lib

### LINKED ISSUE #22062: Store: Remove remaining dependencies on old location lib functions
We have a few remaining dependencies on functions like getCountryData, getStateData and getCountries from woocommerce/lib/countries - let's replace those with calls to the woocommerce/state/sites/locations selectors instead.

This will help us avoid duplicating WooCommerce i18n/locale-info.php code

### PR DESCRIPTION
**Pull request by allendav**
Fixes #22062
See also #21574 

Design:
- Removes what remains of the old US and CA countries lib
- Switches us over to completely rely on the location data from the site itself instead
- Passes the countries as a prop into AddressView
- Make sure all users of AddressView isEditable handle country changes correctly

To Test:

- Delete any products on your test site (important - we still use this to decide whether or not to push defaults for currency, dimensions, etc)
- Optionally clear address and currency related options by installing https://github.com/allendav/woocommerce-reset-currency-and-country (to clear the options, simply visit the page that plugin adds to the Tools menu in wp-admin) 
- Use the WordPress.com console to reset `set_store_address_during_initial_setup` to 0 for your test site
- Navigate to http://calypso.localhost:3000/store/{DOMAIN}
- Set your store to a US address
- Go to /wp-admin/admin.php?page=wc-settings
- Make sure you see the US address, USD, currency pos: left, thousands: "," decimals: "." number of decimals: 2
- Go to /wp-admin/admin.php?page=wc-settings&tab=products
- Make sure you see ounces, inches
- Reset `set_store_address_during_initial_setup` to 0 again

- Set your store to a UK address (no state!)
- Go to /wp-admin/admin.php?page=wc-settings
- Make sure you see the UK address, Pounds Sterling, currency pos: left, thousands: "," decimals: "." number of decimals: 2
- Go to /wp-admin/admin.php?page=wc-settings&tab=products
- Make sure you see kilograms, centimeters
- Reset `set_store_address_during_initial_setup` to 0 again

- Go to Settings > Payments
- Make sure you can switch the store location back and forth between countries that have states and countries that do not
- Go to Settings > Shipping
- Make sure you can switch the address back and forth between countries that have states and countries that do not
- Go to Settings > Taxes
- Make sure that everything renders without error as you move the store location setting
- Go to Order > Edit Order > Customer Billing Address and Customer Shipping Address
- Make sure you can switch the store location back and forth between countries that have states and countries that do not

### CODE REVIEW

**Comment by timmyc** [comment]
Is this good to review once the new `wpcomsh` ships @allendav ?

**Comment by allendav** [comment]
@timmyc wrote:

> Is this good to review once the new wpcomsh ships @allendav ?

Yep! Will have this ready to review today

File: client/extensions/woocommerce/app/dashboard/setup/store-location.js [hunk 1]
```diff
@@ -290,10 +314,12 @@ function mapStateToProps( state, ownProps ) {
 	const settingsGeneralLoaded = areSettingsGeneralLoaded( state, siteId );
 	const storeLocation = getStoreLocation( state, siteId );
 	const locationsLoaded = areLocationsLoaded( state, siteId );
+	const countries = getAllCountries( state, siteId );
 	const countriesWithStates = getCountriesWithStates( state, siteId );
 
 	return {
 		contactDetails,
+		countries,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/dashboard/setup/store-location.js]: Looks like this needs to be added to `propTypes`

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 2]
```diff
@@ -77,7 +95,11 @@ class OrderCustomerInfo extends Component {
 	};
 
 	renderBilling = ( address = {} ) => {
-		const { translate } = this.props;
+		const { countries, loadedLocations, translate } = this.props;
+		if ( ! loadedLocations ) {
+			return null;
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: Do we want to show some sort of loading indicator while the countries are loading?

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 3]
```diff
@@ -173,10 +199,15 @@ export default connect(
 		const siteId = getSelectedSiteId( state );
 		const order = getOrderWithEdits( state );
 
+		const loadedLocations = areLocationsLoaded( state, siteId );
+		const countries = getAllCountries( state, siteId );
+
 		return {
+			countries,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: These two new props could also be added to `propTypes` for this component.

File: client/extensions/woocommerce/app/order/order-customer/dialog.js [hunk 4]
```diff
@@ -75,7 +76,11 @@ class CustomerAddressDialog extends Component {
 	state = {};
 
 	componentDidMount() {
+		const { siteId } = this.props;
 		this.initializeState();
+		if ( siteId && ! this.props.areLocationsLoaded ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/dialog.js]: I liked your `maybeFetchLocations` in the component above, could be used here also.

File: client/extensions/woocommerce/app/order/order-customer/dialog.js [hunk 5]
```diff
@@ -280,13 +296,17 @@ class CustomerAddressDialog extends Component {
 export default connect(
 	state => {
 		const address = getStoreLocation( state );
+		const locationsLoaded = areLocationsLoaded( state );
 		const areSettingsLoaded = areSettingsGeneralLoaded( state );
+		const countries = getAllCountries( state );
 
 		return {
+			areLocationsLoaded: locationsLoaded,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/dialog.js]: `propTypes` :)

File: client/extensions/woocommerce/app/order/order-customer/index.js [hunk 6]
```diff
@@ -70,21 +89,23 @@ class OrderCustomerInfo extends Component {
...
 				updateAddress={ this.updateAddress( 'shipping' ) }
 			/>,
 		];
 	};
 
 	render() {
-		const { isEditing, order, translate } = this.props;
-		if ( ! order ) {
+		const { countries, isEditing, loadedLocations, order, translate } = this.props;
+		if ( ! order || ! loadedLocations ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/index.js]: NM about my prior question about showing a placeholder - I'm seeing a pattern that was already established here of just returning null.

File: client/extensions/woocommerce/app/order/order-customer/index.js [hunk 7]
```diff
@@ -157,11 +178,16 @@ export default connect(
 		const isEditing = isCurrentlyEditingOrder( state );
 		const order = isEditing ? getOrderWithEdits( state ) : getOrder( state, props.orderId );
 
+		const loadedLocations = areLocationsLoaded( state, siteId );
+		const countries = getAllCountries( state, siteId );
+
 		return {
+			countries,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-customer/index.js]: p'typez

File: client/extensions/woocommerce/components/address-view/index.js [hunk 8]
```diff
@@ -33,9 +31,20 @@ class AddressView extends Component {
 			country: PropTypes.string.isRequired,
 			postcode: PropTypes.string,
 		} ),
+		countries: PropTypes.arrayOf(
```
Comment (timmyc) [inline client/extensions/woocommerce/components/address-view/index.js]: 🌟

File: client/extensions/woocommerce/components/form-location-select/states.js [hunk 9]
```diff
@@ -75,7 +75,7 @@ class FormStateSelectFromApi extends Component {
 	renderDisabled = () => {
 		const { translate } = this.props;
 		return (
-			<FormSelect disabled>
+			<FormSelect autoComplete="off" disabled>
```
Comment (timmyc) [inline client/extensions/woocommerce/components/form-location-select/states.js]: curious why the autoComplete off here?

**Comment by timmyc** [comment]
Everything tested out as detailed above. Nice! Should we get one more set of eyes on this since it is a larger PR?

File: client/extensions/woocommerce/components/form-location-select/states.js (hunk 9 above)
Comment (allendav) [reply client/extensions/woocommerce/components/form-location-select/states.js]: It avoids a Chrome warning about auto-complete and disabled

File: client/extensions/woocommerce/app/order/order-customer/index.js (hunk 7 above)
Comment (allendav) [reply client/extensions/woocommerce/app/order/order-customer/index.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/22838/commits/705b284462b5390c6fa399f6c7eaef08f0cc96a9

File: client/extensions/woocommerce/app/order/order-customer/dialog.js (hunk 5 above)
Comment (allendav) [reply client/extensions/woocommerce/app/order/order-customer/dialog.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/22838/commits/705b284462b5390c6fa399f6c7eaef08f0cc96a9

File: client/extensions/woocommerce/app/order/order-customer/create.js (hunk 3 above)
Comment (allendav) [reply client/extensions/woocommerce/app/order/order-customer/create.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/22838/commits/705b284462b5390c6fa399f6c7eaef08f0cc96a9

File: client/extensions/woocommerce/app/dashboard/setup/store-location.js (hunk 1 above)
Comment (allendav) [reply client/extensions/woocommerce/app/dashboard/setup/store-location.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/22838/commits/705b284462b5390c6fa399f6c7eaef08f0cc96a9

**Pull request merged**